source/extensions/filters/http/grpc_http1_reverse_bridge_transcoder/http_methods.cc

```diff
@@ -0,0 +1,40 @@
+#include <algorithm>
+#include <cctype>
+#include <map>
+
+#include "http_methods.h"
+
+namespace Envoy::Extensions::HttpFilters::GrpcHttp1ReverseBridgeTranscoder {
+
+absl::StatusOr<HttpMethod> httpMethodFrom(std::string_view maybe_http_method) {
+  static const std::map<std::string, HttpMethod> mapping{
+      {"GET", HttpMethod::GET},
+      {"POST", HttpMethod::POST},
+      {"PUT", HttpMethod::PUT},
+      {"DELETE", HttpMethod::DELETE},
+  };
+
+  std::string normalized{maybe_http_method};
+  std::transform(normalized.begin(), normalized.end(), normalized.begin(),
+                 [](auto ch) { return std::toupper(ch); });
```

Comment:
Is this safe to do? Write at the beginning of the container that is under transformation?